src/fides/api/models/privacy_preference.py

```diff
@@ -146,6 +197,18 @@ def provided_identity_id(cls) -> Column:
 
     user_geography = Column(String, index=True)
 
+    vendor = Column(
+        String, index=True
+    )  # When saving privacy preferences with respect to a vendor directly
```

Comment:
The name and meaning of this field is something I want to clarify earlier rather than later. I've been assuming the whole time that preferences would be saved against something like sendgrid's vendor id instead of a system fides key, given that multiple systems can point to the same vendor.

https://ethyca.slack.com/archives/C05E74TR0BB/p1692371205716419?thread_ts=1692369966.244759&cid=C05E74TR0BB

But it sounds like system fides key is on the table instead, I just want to make sure we're all on the same page here - this matters a lot for how we aggregate the results and causes potential confusing edge cases.

Reply:
In the followup, we'll need to collapse vendor information together in the overlay, but also support saving preferences against systems that don't have a specific vendor. Adam suggested an additional column, so we have one for vendor id and one for system id.  

For retrieving the last saved preference (in a separate table for easier retrieval across time, tcf versions, etc, multiple identities of the user, etc.) my early thoughts are that we should save against the vendor id *only* in that table if it exists, otherwise, the system id. This might solve the unique constraint issue.